redcap_bridge/utils.py

```diff
@@ -58,4 +61,37 @@ def get_repo_state(path):
         commit_hash = ''
         clean = False
 
-    return commit_hash, clean
\ No newline at end of file
+    return commit_hash, clean
+
+
+def compressed_record(csv_file, compressed_file=None):
+    custom_csv = pd.read_csv(csv_file)
+    df = pd.DataFrame(custom_csv)
+    compressed_file = df.filter(regex='.___.')
+    compressed_file.replace(0,'', inplace=True)
+    for column in compressed_file:
+        if 1 in compressed_file[column].values:
+            style_moda = re.search('___(.+?)', column).group(1)
+            compressed_file[column].replace(1, style_moda, inplace=True)
+        else:
+            compressed_file.pop(column)
+    list_column = compressed_file.columns.tolist()
+
+    for i, item in enumerate(list_column):
```

Comment:
This is still a non-optimal use of `agg`. I think you should be able to only call `.agg` once for each `name`. E.g. with something like this 
```
names = set([c.split('___')[0] for c in compressed_file.columns])
final_df = pd.DataFrame()  # create an empty df to collect all final columns (1 per name)
for name in names:
    final_df.assign(name= compressed_file.filter(f'{name}___.').agg(','.join, axis=1)
return final_df
```